debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/MessageDecoder.java

```diff
@@ -32,10 +32,30 @@
 
     /**
      * Allows MessageDecoder to configure options with which the replication stream is started.
+     * The messages CAN contain type metadata.
      * See PostgreSQL command START_REPLICATION SLOT for more details.
      *
      * @param builder
      * @return the builder instance
      */
-    ChainedLogicalStreamBuilder options(final ChainedLogicalStreamBuilder builder);
+    ChainedLogicalStreamBuilder optionsWithMetadata(final ChainedLogicalStreamBuilder builder);
+
+    /**
+     * Allows MessageDecoder to configure options with which the replication stream is started.
+     * The messages MUST NOT contain type metadata.
+     * See PostgreSQL command START_REPLICATION SLOT for more details.
+     *
+     * @param builder
+     * @return the builder instance
+     */
+    ChainedLogicalStreamBuilder optionsWithoutMetadata(final ChainedLogicalStreamBuilder builder);
+
+    /**
+     * Signals to MessageDecoder that the plug-in may or must not contain type metadata.
+     * 
+     * @param flag - <code>true</code> if messages CAN contain type metadata,
+     * <code>false</code> if messages must not contain type metadata 
+     */
+    default void canHaveMetadata(boolean flag) {
```

Comment:
What's the purpose of this? Can't each plug-in just react to the presence or absence of the metadata? This already seems to be the case for the wal2json plugin. If it's needed indeed, we should use a method name which doesn't sound like a query ("can") but rather like a command, e.g. `setMayContainMetadata()`. Or actually `setContainsMeta()`? Don't we know whether messages have metadata or they don't?

Reply:
This is a little bit tricky part to explain. If we want to get nullbale metadata from `wal2json` plug-in we need to include `include-not-null` fro starting replication. Unfortunately we must think of two versions of the plugin: with metadata that we force upon our users (https://github.com/eulerto/wal2json/commit/f68cb0096c669a2ee5a2d32b54a535513e3cb23b) and without metadata that is deployed on RDS right now. If we use the aforementioned parameter on plug-in the does not support it then the replication start will fail.
Thus I am trying to start the replication with the parameter and if it fails then start it without. If it starts without then I set the flag that albeit the metadata could be present in the message it won't.
As soon as RDS move after the commit in question this code can and will be completely eliminated.
I've renamed the method to `setMayContainMetadata`.